Fix logical error for DynDNS (#949)
jamulussoftware/jamulussoftware.github.io: wiki/en/Unregistered-Servers.md

### Dynamic DNS and why you will probably need it

> Unless your ISP provides you with a fixed IP address, you may find that your address changes over time. You might want to set up "dynamic DNS" to get a static address you can share with others. A "dynamic DNS" provider can supply you with a domain name that you can share, but you need to keep refreshing your dynamic IP with the provider. Your router might support certain providers to do this. If this is not the case, the provider will have instructions on how to set up a dynamic DNS client.

Reviewer: I haven't done this (in a very long time). The dynamic DNS provider gives you an IP address that maps to another IP address? They don't just give you a URL that changes its IP as yours changes?

Author: > The dynamic DNS provider gives you an IP address that maps to another IP address

No. It gives you a (sub) domain.

Reviewer 2: Yeah, a DDNS "provider" is just a hostname registrar.  They'll have a bunch of top level domains, like "is-really-cool.com" or whatever.  You buy the subdomain name, e.g. "ice-cream".  Every so often, you ping them so they pick up your current IP address and the DNS domain name you've purchased from them will then resolve to that IP when anyone does a DNS look up on "ice-cream.is-really-cool.com" (hence using the short TTL).

It's a bit like with a Jamulus Directory, really...

(Note, the Jamulus Directory stores the server name as provided - e.g. the "ice-cream.is-really-cool.com".  Hence it's better to have a name than an IP address which, if your IP provider decides your address has changed, will stop working.  What I can't remember is how that "as provided" works, currently...)

> Unless your ISP provides you with a fixed IP address, you may find that your address changes over time. You might want to set up "dynamic DNS" to get a static address you can share with others. A "dynamic DNS" provider can supply you with a domain name that you can share, but you need to keep refreshing your dynamic IP with the provider. Your router might support certain providers to do this. If this is not the case, the provider will have instructions on how to set up a dynamic DNS client.

Reviewer: ```suggestion
Unless your ISP provides you with a fixed IP address, you may find that your address changes over time. You might want to set up "dynamic DNS" to get a fixed address you can share with others. A "dynamic DNS" provider can supply you with a domain name that you can share, but you need to keep refreshing your dynamic IP with the provider. Your router might support certain providers to do this. If this is not the case, the provider will have instructions on how to set up a dynamic DNS client.
```

Reviewer 2: As I said, it's redundant usage.

Reviewer: Use the **Add a suggestion** interface.

Reviewer 2: It doesn't need a change to what's in the commit.

> Unless your ISP provides you with a fixed IP address, you may find that your address changes over time. You might want to set up "dynamic DNS" to get a static address you can share with others. A "dynamic DNS" provider can supply you with a domain name that you can share, but you need to keep refreshing your dynamic IP with the provider. Your router might support certain providers to do this. If this is not the case, the provider will have instructions on how to set up a dynamic DNS client.

Reviewer 2: ```suggestion
Unless your ISP provides you with a fixed IP address, you may find that your address changes over time. You might want to associate your IP address with a domain name you can share with others that uses "dynamic DNS" (DDNS). A "dynamic DNS" provider can supply you with the domain name and you keep refreshing your IP address with the provider. Your router might support certain DDNS providers to do this. If this is not the case, the provider will have instructions on how to set up a dynamic DNS client.
```

> Unless your ISP provides you with a fixed IP address, you may find that your address changes over time. You might want to set up "dynamic DNS" to get a static address you can share with others. A "dynamic DNS" provider can supply you with a static domain name that you can share, but you need to keep refreshing your dynamic IP with the provider. Your router might support certain providers to do this. If this is not the case, the provider will have instructions on how to set up a dynamic DNS client.

Reviewer 2: "static domain name"?  I'm not familiar with the term.

[From "What is Dynamic DNS"](https://www.checkpoint.com/cyber-hub/cyber-security/what-is-dns-domain-name-server/what-is-dynamic-dns-ddns/)
> The Domain Name Service (DNS) maps hostnames to IP addresses. Dynamic DNS (DDNS) services automatically update their records as IP addresses change to ensure that clients requesting the record for a hostname always receive the correct IP address.

The _domain name_ is _fixed_ (as domain name always are) whilst the IP address is either _static_ or _dynamic_.  "Dynamic DNS" means using a short TTL (time to live) on the DNS A rec entry, so that the IP address for the name is checked often and, if the IP address for that name has changed, then the new IP address will be picked up.

Just delete the word "static" from "static domain name"...

Author: True. That's better.
